whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java

```diff
@@ -1773,6 +1788,215 @@ public void search_inetnum_no_sources_given() {
+        assertThat(whoisResources.getWhoisObjects().get(0).getObjectInfoMessages().get(0), is("" +
+                "% Warning: this route object conflicts with an overlapping RPKI ROA with a different origin AS6505.\n" +
+                "% As a result an announcement for this prefix may be rejected by many autonomous systems. You should" +
+                " either remove this route: object or delete the ROA.\n"));
+    }
+
+    @Test
+    public void search_route_roa_mismatch_validation_enabled_as_xml() {
+        databaseHelper.addObject(RpslObject.parse("" +
+                "route:           193.4.0.0/16\n" +
+                "descr:           Ripe test allocation\n" +
+                "origin:          AS102\n" +
+                "admin-c:         TP1-TEST\n" +
+                "mnt-by:          OWNER-MNT\n" +
+                "mnt-lower:       OWNER-MNT\n" +
+                "source:          TEST-NONAUTH\n"));
+        ipTreeUpdater.rebuild();
+
+        rpkiDataProvider.setRoas(Lists.newArrayList(
+                new Roa(6505, 16, "193.4.0.0/16", ARIN)
+        ));
+
+        final WhoisResources whoisResources = RestTest.target(getPort(), "whois/search.xml?query-string=193.4.0.0/16AS102&flags=roa-validation&flags=no-referenced")
+                .request()
+                .get(WhoisResources.class);
+
+        assertThat(whoisResources.getWhoisObjects(), hasSize(1));
+
+        assertThat(whoisResources.getWhoisObjects().get(0).getObjectInfoMessages(), hasSize(1));
+        assertThat(whoisResources.getWhoisObjects().get(0).getObjectInfoMessages().get(0), is("" +
+                "% Warning: this route object conflicts with an overlapping RPKI ROA with a different origin AS6505.\n" +
+                "% As a result an announcement for this prefix may be rejected by many autonomous systems. You should" +
+                " either remove this route: object or delete the ROA.\n"));
+    }
+
+    @Test
+    public void search_route_roa_mismatch_validation_enabled_as_xml_strings() {
+        databaseHelper.addObject(RpslObject.parse("" +
+                "route:           193.4.0.0/16\n" +
+                "descr:           Ripe test allocation\n" +
+                "origin:          AS102\n" +
+                "admin-c:         TP1-TEST\n" +
+                "mnt-by:          OWNER-MNT\n" +
+                "mnt-lower:       OWNER-MNT\n" +
+                "source:          TEST-NONAUTH\n"));
+        ipTreeUpdater.rebuild();
+
+        rpkiDataProvider.setRoas(Lists.newArrayList(
+                new Roa(6505, 16, "193.4.0.0/16", ARIN)
+        ));
+
+        final String whoisResources = RestTest.target(getPort(), "whois/search.xml?query-string=193.4.0" +
+                        ".0/16AS102&flags=roa-validation&flags=no-referenced")
+                .request()
+                .get(String.class);
+
+        assertThat(whoisResources, is("" +
```

Comment:
can you assert for WArning? Much more readble to see you are checking for messages in final xml. This test case is for roa mismatch only

Reply:
I really want this test here, we are doing this in another test cases as well. 
There is another test doing what you are saying "search_route_roa_mismatch_validation_enabled_as_xml"